### PR TITLE
[AKS] re-apply the changes to making the count field optional to fix regressions

### PR DESCRIPTION
**Pull request by marwanad**
PR: https://github.com/Azure/azure-rest-api-specs/pull/7746 introduced changes to make the `count` field optional. This PR hot-fixed prior API versions and was merged on **November 17th**. 

The [base](https://github.com/gtracer/azure-rest-api-specs/commit/fd855a5f887caaa675fdbb18d0b4dcebe1fa25fa) for the AKS November API was created on **November 15th** which means it missed this change.

The consequence is that all API versions starting from November didn't include the change in #7746.

This fix PR applies the changes in #7746 to all API versions starting from November and above. This should qualify as a bug fix and not a breaking change.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by marwanad** [comment]
@tombuildsstuff fyi for the Terraform provider. I'm not sure what Go SDK version and AKS API version the provider uses but just in case an update is needed.

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by xizhamsft** [comment]
LGTM

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 1]
```diff
@@ -1426,10 +1429,7 @@
         "count": {
           "type": "integer",
           "format": "int32",
-          "maximum": 100,
-          "minimum": 0,
-          "description": "Number of agents (VMs) to host docker containers. Allowed values must be in the range of 0 to 100 (inclusive). The default value is 1. ",
-          "default": 1
```
Comment (NullMDR) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:1432]: Removing default causes breaking change. Can we keep this?

**Review by PhoenixHe-NV — CHANGES_REQUESTED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json (hunk 1 above)
Comment (marwanad) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:1432]: @NullMDR per the PR description, those defaults shouldn't have been there in the first place. 

https://github.com/Azure/azure-rest-api-specs/pull/7746 removed those defaults but when we branched off for November API, we didn't include those PR.

Comment (marwanad) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:1432]: The changes here are just a cherry-pick of #7746 which is what the state of the spec should be.

**Comment by yungezz** [comment]
hi @NullMDR, this "breaking change" actually is fixing Swagger issue which is wrongly there. Could you pls help to review it?  thanks.

**Comment by NullMDR** [comment]
@akning-ms Please help to merge this PR with breaking change approved

**Pull request merged**